iwyu_tool_test.py

```diff
@@ -208,6 +259,40 @@ def test_consecutive_spaces(self):
         self.assert_win_split('clang  -I. \t     -A',
                               ['clang', '-I.', '-A'])
 
+class BootstrapTests(unittest.TestCase):
+    def setUp(self):
+        self.iwyu_tool_main_mock =  MockIwyuToolMain()
+        self.sys_exit_stub = StubSysExit()
+        self.sys_exit_stub.will_return(0)
+
+    def tearDown(self):
+        self.iwyu_tool_main_mock.reset()
+        self.sys_exit_stub.reset()
+
+    def test_argument_parser_sets_argument_correctly(self):
+        """ Check that arguments are verbatim injected to iwyu. """
+        argv_stub = StubSysArgv(["iwyu_tool.py", "-p", ".", "--", "arg1"])
+        iwyu_tool._bootstrap()
+        iwyu_call_args = self.iwyu_tool_main_mock.get_call_args()
+        self.assertIn(["arg1"], iwyu_call_args)
+
+    def test_argument_parser_does_include_xiwyu_flag(self):
+        """ Check that -Xiwyu and --iwyu_opt arguments are passed to the iwyu call. """
```

Comment:
`--iwyu_opt` has no special meaning for IWYU, so it probably doesn't need to be called out in the test here. The important quality is that `-Xiwyu --something` is forwarded.

Reply:
Done ✔️ 
Absolutely right. I misunderstood the meaning of 'iwyu_opt' flag.